azure/scope/machine.go

```diff
@@ -532,3 +538,26 @@ func (m *MachineScope) GetVMImage() (*infrav1.Image, error) {
 	m.Info("No image specified for machine, using default Linux Image", "machine", m.AzureMachine.GetName())
 	return azure.GetDefaultUbuntuImage(to.String(m.Machine.Spec.Version))
 }
+
+// SetSubnetName defaults the AzureMachine subnet name defined when empty to the name of the subnet with role 'node' where there is only one of them.
+// Note: this logic exists only for purposes of ensuring backwards compatibility for old clusters created without the `subnetName` field being set, and should be removed in the future.
```

Comment:
IIUC, this also handles setting subnet name for control plane nodes. Unless, we are planning to implement multiple subnets for cp nodes, this is cannot be removed, I think.

Reply:
The plan would be to make the `subnetName` field to be mandatory. Once that happens, this logic is not needed anymore. The same applies for both `control-plane` and `node` subnets.